src/portal-impl.c

```diff
@@ -316,65 +369,48 @@ load_portal_configuration (gboolean opt_verbose)
   if (portal_dir == NULL)
     portal_dir = DATADIR "/xdg-desktop-portal";
 
-  conf = g_new0 (PortalConfig, 1);
+  user_portal_dir = g_build_filename (g_get_user_config_dir (),
+                                      "xdg-desktop-portal",
+                                      NULL);
 
   desktops = get_current_lowercase_desktops ();
   for (size_t i = 0; desktops[i] != NULL; i++)
     {
       g_autofree char *portals_conf = g_strdup_printf ("%s-portals.conf", desktops[i]);
-      g_autofree char *path = g_build_filename (portal_dir, portals_conf, NULL);
-      g_auto(GStrv) ifaces;
-
-      g_autoptr(GKeyFile) key_file = g_key_file_new ();
-
-      g_debug ("Looking for portals configuration in '%s'", path);
-      if (!g_key_file_load_from_file (key_file, path, G_KEY_FILE_NONE, NULL))
-        continue;
 
-      if (opt_verbose)
-        g_debug ("Loading portal configuration file '%s' for desktop '%s'",
-                 path,
-                 desktops[i]);
-
-      ifaces = g_key_file_get_keys (key_file, "preferred", NULL, NULL);
+      conf = load_portal_configuration_for_dir (opt_verbose, portal_dir, portals_conf);
+      if (conf != NULL)
+        {
+          if (opt_verbose)
+            g_debug ("Using system portal configuration file '%s' for desktop '%s'",
+                     portals_conf,
+                     desktops[i]);
 
-      g_autoptr(GPtrArray) interfaces = g_ptr_array_new_full (g_strv_length (ifaces) + 1, NULL);
+          config = g_steal_pointer (&conf);
+          return;
+        }
 
-      if (ifaces != NULL)
+      conf = load_portal_configuration_for_dir (opt_verbose, user_portal_dir, portals_conf);
```

Comment:
Isn't this the wrong way around? Normally user configuration overrides system configuration while here system configuration takes precedence.

Reply:
Indeed, I left this in because of the test suite and never rolled it back.

There's also the option of merging multiple layers of configuration, with the last one winning, but I'll leave that for another pull request.

Reply:
Fixed.